chain/network/src/peer/peer_actor.rs

```diff
@@ -1424,6 +1434,13 @@ impl actix::Actor for PeerActor {
                     self.stream_id,
                     self.closing_reason.clone().unwrap_or(ClosingReason::Unknown),
                 );
+
+                if network_state.is_recent_outbound_connection(&conn.peer_info.id) {
+                    network_state.set_reconnect_attempts(
+                        &conn.peer_info,
+                        RECENT_OUTBOUND_CONNECTION_MAX_RECONNECT_ATTEMPTS,
+                    );
+                }
```

Comment:
IMO this logic should reside either in network_state, or even deeper: in PeerStore